services/venice-router/src/main/java/com/linkedin/venice/router/acl/RouterStoreAclHandler.java

```diff
@@ -0,0 +1,83 @@
+package com.linkedin.venice.router.acl;
+
+import static com.linkedin.venice.router.api.RouterResourceType.TYPE_INVALID;
+
+import com.linkedin.venice.acl.AclCreationDeletionListener;
+import com.linkedin.venice.acl.DynamicAccessController;
+import com.linkedin.venice.acl.handler.AbstractStoreAclHandler;
+import com.linkedin.venice.authorization.IdentityParser;
+import com.linkedin.venice.exceptions.VeniceUnsupportedOperationException;
+import com.linkedin.venice.meta.ReadOnlyStoreRepository;
+import com.linkedin.venice.router.api.RouterResourceType;
+import io.netty.channel.ChannelHandler;
+
+
+/**
+ * Store-level access control handler, which is being used by both Router and Server.
+ */
+@ChannelHandler.Sharable
+public class RouterStoreAclHandler extends AbstractStoreAclHandler<RouterResourceType> {
+  public RouterStoreAclHandler(
+      IdentityParser identityParser,
+      DynamicAccessController accessController,
+      ReadOnlyStoreRepository metadataRepository) {
+    super(identityParser, accessController, metadataRepository);
+    metadataRepository.registerStoreDataChangedListener(new AclCreationDeletionListener(accessController));
+  }
+
+  @Override
+  protected boolean needsAclValidation(RouterResourceType resourceType) {
+    switch (resourceType) {
+      case TYPE_LEADER_CONTROLLER:
+      case TYPE_LEADER_CONTROLLER_LEGACY:
+      case TYPE_KEY_SCHEMA:
+      case TYPE_VALUE_SCHEMA:
+      case TYPE_LATEST_VALUE_SCHEMA:
+      case TYPE_GET_UPDATE_SCHEMA:
+      case TYPE_ALL_VALUE_SCHEMA_IDS:
+      case TYPE_CLUSTER_DISCOVERY:
+      case TYPE_STREAM_HYBRID_STORE_QUOTA:
+      case TYPE_STREAM_REPROCESSING_HYBRID_STORE_QUOTA:
+      case TYPE_STORE_STATE:
+      case TYPE_PUSH_STATUS:
+      case TYPE_ADMIN: // Access control for Admin operations are handled in AdminOperationsHandler
+      case TYPE_CURRENT_VERSION:
+      case TYPE_RESOURCE_STATE:
+      case TYPE_BLOB_DISCOVERY:
+      case TYPE_REQUEST_TOPIC:
+        return false;
+      case TYPE_STORAGE:
+      case TYPE_COMPUTE:
+        return true;
+      case TYPE_INVALID:
```

Comment:
There is a big list of the switch cases, so if every valid request needs to go through the list, it will be slightly inefficient, right?
If we still to this pattern, I think we should move `TYPE_STORAGE` and `TYPE_COMPUTE` up, which are more latency sensitive with more QPS, and another way is to use a `HashMap`.
WDYT?

Reply:
I can move it around, but some resources I read online say that the compiler will make this into a lookup table on it's own if there are >5 branches

Reply:
https://www.geeksforgeeks.org/switch-vs-else/

Reply:
But I could get rid of the rest, and only keep `TYPE_STORAGE` and `TYPE_COMPUTE`, and let the tests handle checking the case that all types are handled already. We have this test now

Reply:
This is how it would look:
```
  @Override
  protected boolean needsAclValidation(RouterResourceType resourceType) {
    if (resourceType == TYPE_STORAGE || resourceType == TYPE_COMPUTE) {
      return true;
    }

    if (resourceType == TYPE_INVALID) {
      throw new VeniceUnsupportedOperationException(resourceType.name());
    }

    return false;
  }
```

Reply:
Got it.
Only keeping `TYPE_STORAGE` and `TYPE_COMPUTE` sounds good to me.
This would also reduce the maintenance burden when we adding more admin/metadata endpoints in the future.